lib/ansible/modules/source_control/git.py

```diff
@@ -432,8 +450,23 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
                         "HEAD, branches, tags or in combination with refspec.")
     if reference:
         cmd.extend(['--reference', str(reference)])
+    needs_separate_git_dir_fallback = False
+
+    if separate_git_dir:
+        git_version_used = git_version(git_path, module)
+        if git_version_used is None:
+            module.fail_json(msg='Can not find git executable at %s' % git_path)
+        if git_version_used < LooseVersion('1.7.5'):
```

Comment:
It might be a better idea to feature-detect its support using smth like `git clone --help | grep '\-\-separate-git-dir'` (while a bit more expensive).